core/dictgen/src/rootcling_impl.cxx

```diff
@@ -4221,6 +4221,9 @@ int RootClingMain(int argc,
    clingArgs.push_back("-Xclang");
    clingArgs.push_back("-main-file-name");
    clingArgs.push_back("-Xclang");
+   // FIXME: Enable this once we switch cling to use the driver.
+   clingArgs.push_back("-fmodules-embed-all-files");
+   clingArgs.push_back("-Xclang");
    clingArgs.push_back((dictname + ".h").c_str());
```

Comment:
This breaks the parameters. We want `--cc1` to see `-main-file-name dictname.h -fmodules-embed-all-files`; this change makes that `-main-file-name -fmodules-embed-all-files dictname.h` i.e. the main file name is now broken.

Reply:
Sure, fixed.